actionpack/lib/action_controller/metal/strong_parameters.rb

```diff
@@ -231,6 +231,7 @@ class Parameters
     #   params.permitted?  # => true
     #   Person.new(params) # => #<Person id: nil, name: "Francesco">
     def initialize(parameters = {})
+      raise ArgumentError, "invalid parameters given, the parameters should be a hash" unless parameters.is_a?(Hash)
```

Comment:
I think it is better to check for `parameters.respond_to?(:with_indifferent_access)`